torch/distributed/_shard/sharded_tensor/interface.py

```diff
@@ -11,19 +11,29 @@
 
 
 class ShardedTensorInterface(torch.Tensor):
+    """
+    ShardedTensorInterface is an interface of the ShardedTensor abstraction
+    which represents Tensors that are sharded across multiple devices and
+    multiple processes.
+
+    :class:`ShardedTensor` is a concrete implementation of this interface
+    and attached local shards with real data/memory.
```

Comment:
Typically an interface shouldn't refer to its implementations in docs or otherwise. We can add such docs in ShardedTensor itself.